src/stactools/core/projection.py

```diff
@@ -1,9 +1,14 @@
-from copy import deepcopy
-from typing import Any, Dict, List, Optional, Sequence, Union
+import json
+import warnings
+from typing import Any, Dict, List, Optional, Union, cast
 
 import pyproj
 import rasterio.crs
 import rasterio.transform
+from rasterio.warp import transform_geom
+from shapely import Geometry
```

Comment:
I don't think we want `shapely.Geometry` here, that looks like it comes straight from GEOS: https://github.com/shapely/shapely/blob/1ceb6b8be378e14e474fa905575dcad28c2477a6/shapely/__init__.py#L2. Looks like [`transform_geom` wants a GeoJSON-like dict](https://rasterio.readthedocs.io/en/stable/api/rasterio.warp.html#rasterio.warp.transform_geom).

Reply:
> So I guess the type requirements should be that?

Yeah, if you want a protocol (I forget if **stactools** has one) I have example here: https://github.com/gadomski/antimeridian/blob/0fa1ccfc466842abbe8525a41c54f0bd2ab53123/src/antimeridian/_implementation.py#L45-L55.

> Are you ok with the output being a Geometry object?

Yup, worksforme.

Reply:
thanks! I duplicated that protocol over to stactools.core.geometry